lib/connection.js

```diff
@@ -345,6 +345,9 @@ Connection.prototype.abort_socket = function (socket) {
         this.socket.removeAllListeners('data');
         this.socket.removeAllListeners('error');
         this.socket.removeAllListeners('end');
+        if (typeof this.socket.destroy === 'function') {
```

Comment:
[socket.destroy](https://nodejs.org/docs/latest-v12.x/api/net.html#net_socket_destroy_exception) has been available since node 0.1.90 so should be safe to call. I wrapped it in a `typeof` because I wasn't sure how this would play with websockets.